akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -13,52 +13,78 @@ import akka.stream._
 import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
 import akka.util.{ ByteString, OptionVal, PrettyByteString }
 import akka.actor.EmptyLocalActorRef
-import akka.remote.artery.compress.{ InboundCompressions, OutboundCompressions, OutboundCompressionsImpl }
+import akka.remote.artery.compress.InboundCompressions
 import akka.stream.stage.TimerGraphStageLogic
 import java.util.concurrent.TimeUnit
+import scala.concurrent.Future
+import akka.remote.artery.compress.CompressionTable
+import akka.Done
+import akka.stream.stage.GraphStageWithMaterializedValue
+import scala.concurrent.Promise
+
+/**
+ * INTERNAL API
+ */
+private[remote] object Encoder {
+  private[remote] trait ChangeOutboundCompression {
+    def changeActorRefCompression(table: CompressionTable[ActorRef]): Future[Done]
+    def changeClassManifestCompression(table: CompressionTable[String]): Future[Done]
+    def clearCompression(): Future[Done]
+  }
+
+  private[remote] class ChangeOutboundCompressionFailed extends RuntimeException(
+    "Change of outbound compression table was unsuccessful, e.g. because materialization not completed yet")
```

Comment:
was unsuccessful => failed?
because materialization not completed yet => because materialization did not complete yet?